apps/recovery-portal/src/main/webapp/password-recovery-confirm.jsp

```diff
@@ -35,7 +36,12 @@
     if (StringUtils.isBlank(callback)) {
         callback = request.getParameter("redirect_uri");
     }
-    String tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+    String tenantDomain;
+    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+        tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+    } else {
+        tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+    }
```

Comment:
If this logic is there in all jsps, shall we look into separating it to a common one and reuse. Check if such common jsp is already available